packages/core/src/babel-plugin-adjust-imports.ts

```diff
@@ -261,6 +302,15 @@ export default function main() {
         },
       },
       CallExpression(path: NodePath<CallExpression>, state: State) {
+        if (isImportSyncExpression(path) || isDynamicImportExpression(path)) {
+          const [source] = path.get('arguments');
+          let { opts } = state;
+          opts.dynamicImports.push((source.node as any).value);
```

Comment:
Instead of using `opts.dynamicImports` to store this state, I think it would be better to add an `isDynamic: boolean` argument to the `adjustSpecifier()` function. Especially because a specifier can be used both statically and dynamically, and the static locations should still be a static error, only the dynamic locations should compile to a runtime error.